### PR TITLE
KIT-228 - Ajouter un flux RSS

### PR DESCRIPTION
**Pull request by Tylyfox**

### CODE REVIEW

**Review by Newaluigi — APPROVED** [review]

File: src/app/components/rss/rss-feed-all/rss-feed-all.component.ts [hunk 1]
```diff
@@ -1,36 +1,40 @@
...
+    this.getRssData();
+    this.subscribeToRssFeedsUpdated();
+  }
+
+  getRssData(): void {
     this.rssFeedService.getAllRssFeeds().subscribe({
-      next: (response: RssFeed[]): void => {
-        this.rssLink = response.map((rssFeed: RssFeed) => rssFeed.url);
-        this.getRssData();
+      next: (response: any[]): void => {
```
Comment (Benja-san) [inline src/app/components/rss/rss-feed-all/rss-feed-all.component.ts:22]: why any type ?

File: src/app/services/RssFeedService/rss.service.ts [hunk 2]
```diff
@@ -1,24 +1,28 @@
...
-  rssLinks: string[] = [];
-  private apiUrl = 'http://localhost:8080/api/rssFeeds';
+  private rssFeedsUpdatedSubject: Subject<void> = new Subject<void>();
   constructor(private http: HttpClient) {}
-  addRssLink(link: string): void {
-    this.rssLinks.push(link);
-  }
-  getRssLinks(): string[] {
-    return this.rssLinks;
+  addRssLink(url: string): Observable<any> {
```
Comment (Benja-san) [inline src/app/services/RssFeedService/rss.service.ts:15]: avoid any type

**Review by Benja-san — CHANGES_REQUESTED** [review]

**Review by Newaluigi — APPROVED** [review]

File: src/app/pages/rss/rss.component.ts [hunk 3]
```diff
@@ -2,27 +2,43 @@ import { Component } from '@angular/core';
...
     const dialogRef = this.dialog.open(RssModalComponent, {
-      width: '40%',
+      width: '50%',
     });
 
     dialogRef.afterClosed().subscribe((result) => {
       if (result) {
-        this.rssService.addRssLink(result);
+        this.rssService.getRssData(result).subscribe({
+          next: (response: any): void => {
```
Comment (Benja-san) [inline src/app/pages/rss/rss.component.ts:23]: still any type

**Review by Benja-san — CHANGES_REQUESTED** [review]

File: src/app/pages/rss/rss.component.ts [hunk 4]
```diff
@@ -20,7 +21,8 @@ export class RssComponent {
     dialogRef.afterClosed().subscribe((result) => {
       if (result) {
         this.rssService.getRssData(result).subscribe({
-          next: (response: any): void => {
+          next: (response: RssResponse): void => {
+            console.log(response);
```
Comment (Benja-san) [inline src/app/pages/rss/rss.component.ts:25]: remove useless console log 😃

**Review by Benja-san — CHANGES_REQUESTED** [review]

**Review by Benja-san — APPROVED** [review]

**Review by Newaluigi — APPROVED** [review]

**Pull request merged**